pallets/settlement/src/lib.rs

```diff
@@ -1035,12 +1163,18 @@ impl<T: Trait> Module<T> {
                     ));
                 }
                 LegStatus::ExecutionPending => {
-                    // Tokens are unlocked, need to be unlocked
-                    T::Portfolio::unlock_tokens(
-                        &leg_details.from,
-                        &leg_details.asset,
-                        &leg_details.amount,
-                    )?;
+                    // Tokens are unlocked, need to be unlocked.
+                    match leg_details {
+                        Leg::NonConfidentialLeg(_) => {
+                            let (asset, amount) = Self::leg_asset_and_amount(&leg_details)?;
+                            let (from, _) = Self::leg_from_to(&leg_details)?;
+                            T::Portfolio::unlock_tokens(&from, &asset, &amount)?;
+                        }
+                        Leg::ConfidentialLeg(_) => {
+                            return Err(Error::<T>::ConfidentialModeNotSupportedYet.into())
```

Comment:
Since it is possible to have `NonConfidentialLeg` and `ConfidentialLeg` in the instruction, we shouldn't be returning an error here. We should just skip unlocking tokens for `ConfidentialLeg` but still unlock the tokens for `NonConfidentialLeg`.

Reply:
Removed the error.